### PR TITLE
Integrate the ctez wrapper into Checker

### LINKED ISSUE #279: Implement an FA2 wrapper for ctez
Now that we've switched to the tez-wrapper-based design, the biggest remaining obstacle for finishing #213 is dealing with the structural differences between TEZ and FA2 that we've found no way to share between the two build/deployment paths. As far as I can tell, this boils down to two specific issues, interrelating with each other:

1. Price calculation in `touch`:
    - `collateral = TEZ`: KIT/CHF = TEZ/CHF (from index) * CTEZ/TEZ (from ctez) * KIT/CTEZ (from cfmm)
    - `collateral = FA2`: KIT/CHF = FA2/CHF (from index) * KIT/FA2 (from cfmm)

    this issue is discussed here: https://github.com/tezos-checker/checker/issues/213#issuecomment-923758356

2. The token contained in the CFMM compared to the token used as collateral:
    - `collateral = TEZ`: `cfmm.ml` contains ctez **_specifically_**, also distinct from the collateral which is tez_token **_specifically_**
    - `collateral = FA2`: `cfmm.ml` contains the same token as is used for collateral, whichever token that is.

The [comment on #213](https://github.com/tezos-checker/checker/issues/213#issuecomment-923758356) is primarily focused on the first issue, but, while trying to make progress on the the second issue (on [this branch](https://github.com/tezos-checker/checker/compare/gkaracha/cfmm-token)), I came across another structural difference:

3. Token standard used for fund transfer:
    - `collateral = TEZ`: `cfmm.ml` contains ctez and it uses ctez's [FA1.2 API][fa12]
    - `collateral = FA2`: `cfmm.ml` contains the same FA2 that is used as collateral so **_I expect_** it'll use the token's [FA2 API][fa12]

FA1.2 and FA2 are structurally different, so the branching as stated above is quite pervasive (i.e., we'll need conditional compilation or something equivalent to get generalization to work). Ideally we won't have to go that way. However, as far as I can tell, ctez does not offer an FA2 interface. To achieve uniformity in the `%transfer` calls we'd have to always use the FA1.2 standard for the cfmm, but then we require the token used to respect both the FA2 and the FA1.2 standard.

This looks like a blocker for #213.

[fa2]: https://gitlab.com/tezos/tzip/-/blob/master/proposals/tzip-12/tzip-12.md
[fa12]: https://gitlab.com/tezos/tzip/-/blob/master/proposals/tzip-7/tzip-7.md

### PR DESCRIPTION
**Pull request by gkaracha**
Fixes #279 by utilizing the FA2 ctez wrapper introduced in #286 as the de facto contract we use for the CFMM.

### CODE REVIEW

**Comment by gkaracha** [comment]
@dorranh feel free to have a look/merge this **_after_** #289; that one is certainly of higher priority at the moment.

**Comment by gkaracha** [comment]
Also, while developing this I came across some issues due to the way we currently deal with a couple token ids:

* `tokens.collateral.token_id` (in `checker.yaml`) and `TezWrapper.tez_token_id` must be the same when `collateral = TEZ`.
* `tokens.cfmm_token.token_id` (in `checker.yaml`) and `Wctez.wctez_token_id` must be the same when `collateral = TEZ`.
Currently `TezWrapper.tez_token_id` and `Wctez.wctez_token_id` are set manually which is a problem (when they get out of sync Checker gets into an inconsistent state and the e2e tests fail).

File: src/checker.ml [hunk 1]
```diff
@@ -589,16 +597,20 @@ let entrypoint_add_liquidity (state, p: checker * (ctok * kit * lqt * Ligo.times
...
+      ];
     } in
   let op =
-    LigoOp.Tezos.fa12_transfer_transaction
-      transfer
-      (Ligo.tez_from_literal "0mutez")
-      (get_transfer_ctok_fa12_entrypoint state.external_contracts) in
+    LigoOp.Tezos.fa2_transfer_transaction
+      [transfer] (Ligo.tez_from_literal "0mutez")
+      (get_transfer_ctok_fa2_entrypoint state.external_contracts) in
```
Comment (dorranh) [inline src/checker.ml:613]: Random, but the diff of these changes from FA1.2 to FA2 transfers looks really nice and tidy.

**Review by dorranh — APPROVED** [review]
Very nice, LGTM!

**Comment by gkaracha** [comment]
Great, thanks for having a look! :pray:

**Pull request merged**